### ISSUE TITLE
How to use StatsAggregator (esp. to collect prefetch size)

### ISSUE THREAD

**Issue by Flamefire** [tensorflow/tensorflow#48500]
## URL(s) with the issue:

https://www.tensorflow.org/api_docs/python/tf/data/experimental/StatsAggregator

## Description of issue (what needs changing):

The documentation does not fully describe how to use `StatsAggregator`, i.e. it is missing what the output should be, where it is and what to do with it.

### Clear description

Backstory: I had a setup, where  speeding up the training step did not yield a speedup of the training which turns out to be dataset input pipeline performance and could be solved by `prefetch(large_number)`
However using the autotune parameter did not help here and I wanted to know why.
In the source code I've seen, that there is a possibility to collect the current prefetch buffer size over training steps so I wanted to look what the chosen values are and how they change, i.e. check if autotune is working as expected
I seems it is not increasing the buffer size fast enough or at all.

However I was not able to figure out how to get this information. And when using `StatsAggregator` it shows a deprecation warning and refers to TF Profiler but the information I need is not there at all. 

### Usage example

Is there a usage example?

No

**Comment by aaudiber** [comment]
Regarding prefetch buffer size autotune debugging, first figure out whether or not you are using legacy autotune. This will show up in the TF Profiler trace of the Prefetch op. If you are using legacy autotune, the behavior is described [here](https://github.com/tensorflow/tensorflow/blob/5dcfc51118817f27fad5246812d83e5dccdc5f72/tensorflow/core/kernels/data/prefetch_autotuner.h#L24-L39). Otherwise, the buffer size is tuned in [model.cc](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/framework/model.cc#L116).

I'm not familiar with StatsAggregator. Given that it's deprecated, I don't think we will invest effort into improving its documentation. TF Profiler gives much more detailed information for performance debugging.

**Comment by Flamefire** [comment]
Thanks for the reply. I've seen the term "legacy autotune" in the code but I haven't been able to find anything of it in the TF profiler. Where would I have to look for that and how can I influence which autotune is used?

In the "trace viewer" tool I see this when selecting the Prefetch op:

```
Args |  
-- | --
buffer_limit | "3"
id | "1886018925671364970"
long_name | "Iterator::Model::MaxIntraOpParallelism::Prefetch"
parent_id | "3180932205155061847"
```

When running with `TF_CPP_MIN_LOG_LEVEL=0 TF_CPP_MIN_VLOG_LEVEL=2` I don't see any "tunable" output related to prefetch, so I'd guess it's the legacy one?

I think it would be very useful to have a graph printing the chosen values of the buffer sizes over the step number and (as I've seen some code related to the StatsAggregator) the latency of dataset operations (i.e. time from call to return) over the step number but I wasn't able to find anything like that in the Profile tab.

The closest seems to be the trace viewer, but even for only 3 local GPUs there are about 2 dozens of `tf_data_*_iterator_resource` timelines which I find very hard to interpret.

And the `tf_data_bottleneck_analysis` shows my 3 devices + 1 host with 32 ranks each with something which could be useful if it wasn't split in (3+1)*32=128 pipeline views which I couldn't find an explanation for how they map to my defined pipeline or to interpret those "ranks".

So bottom line: Where do I find that information in TF Profile which seemingly is/was available only through StatsAggregator?

**Comment by aaudiber** [comment]
In TF 2.4 the trace viewer will show information about autotuning as well as the current buffer size. Is it possible to update to a newer version?

**Comment by Flamefire** [comment]
I am using TF 2.4.1. Where do I have to look for that information?
The only place I've seen is when selecting an op/event in the trace viewer but that is unsuitable to see how the buffer size (e.g.) changes, as selecting dozens or hundreds of such events and manually check for changes is unfeasible.
